pkg/service/service.go

```diff
@@ -599,8 +616,11 @@ func (s *Service) upsertService(params *lb.SVC) (bool, lb.ID, error) {
 	}
 
 	if l7lbInfo != nil && l7lbInfo.envoyBackendRefs != nil && s.envoyCache != nil {
-		// Upsert backends as Envoy endpoints
-		if err = s.envoyCache.UpsertEnvoyEndpoints(name, svc.backends); err != nil {
+		// Filter backend based on list of port numbers, then upsert backends
+		// as Envoy endpoints
+		be := filterServiceBackends(svc, l7lbInfo.frontendPorts)
```

Comment:
This code should live in a different package outside of `pkg/service`.

Reply:
agreed, I will tackle this change along with other related changes post 1.12.

https://github.com/cilium/cilium/pull/19647#discussion_r883478969